velox/exec/fuzzer/AggregationFuzzer.cpp

```diff
@@ -386,14 +394,17 @@ void AggregationFuzzer::go() {
         auto sortingKeys = generateSortingKeys("s", argNames, argTypes);
         auto input = generateInputDataWithRowNumber(
             argNames, argTypes, partitionKeys, signature);
+        auto [convertedInput, projections] =
+            referenceQueryRunner_->inputProjections(input);
 
-        logVectors(input);
+        logVectors(convertedInput);
 
         bool failed = verifyWindow(
             partitionKeys,
             sortingKeys,
             call,
-            input,
+            convertedInput,
+            projections,
```

Comment:
Do we need to adopt this projection during Velox executions? I wonder whether we can just do the transformation in the SQL to be sent to Presto? Or would we miss some test coverage if we do that?

Reply:
I wanted to keep the plans as close as possible to ensure we're doing an apples to apples comparison. It could more uncertainty when debugging failures otherwise (is a difference in behavior because the plans are different or because of an issue in Velox?).

This also gives us the freedom to not produce the exact same data from the projections as the VectorFuzzer produced.  E.g. suppose the type wasn't perfectly inversible, i.e. f1(f2(x)) <> x all the time, we can still use f2 to convert the intermediate type to an input type and f1 to convert it back (it still produces random data, though maybe missing a subset of values which is better than not testing the type at all).